### DISCUSSION TITLE
Add Hive as a producer?

### DISCUSSION THREAD

**Discussion by functicons** [OpenLineage/OpenLineage#2262] [General]
I'm an Apache Hive user and new to the OpenLineage project. In the doc, I noticed that there is no support for Hive as a producer. I'm wondering about the reason and is there any plan to add it? Thanks!

**Comment by mobuchowski** [comment]
Hey @functicons. There's no particular reason, beyond that nobody worked on supporting Hive yet. 

If you're willing to give a shot implementing Hive integration, I can help you on the OpenLineage side - making sure it produces conformant events.

**Comment by functicons** [comment]
Hey @mobuchowski, thanks for your quick and clear answer! That helped me understand the reason.

Yes, I'm interested in implementing Hive integration. Could you share some pointers? Requirements, patterns and examples of implementing a new producer, etc, all are appreciated.

Thanks!

**Reply by mobuchowski** [reply]
@functicons sorry for not replying, the notification got lost to me 😞 

From Hive side, I'd recommend going to Hive community to understand best how to write plugin or otherwise integrate with it. I'm very far from Hive expert and cannot give any good recommendation for it. However, the information you want to get is  

From OpenLineage side, you'd want to use OpenLineage client https://openlineage.io/docs/client/java/ to emit the events. I would focus on getting Dataset naming right, as this mean the integration would work well with other ones.
